mtags/src/main/scala/scala/meta/internal/pc/Completions.scala

```diff
@@ -1632,6 +1675,93 @@ trait Completions { this: MetalsGlobal =>
+     *
+     * @param editRange the range in the original source file.
+     * @param associatedDef the memberDef associated with the scaladoc to complete.
+     *                      This class will construct scaladoc based on the params of this definition.
+     */
+    case class Scaladoc(
+        editRange: l.Range,
+        associatedDef: MemberDef
+    ) extends CompletionPosition {
+      override def contribute: List[Member] = {
+        val lines = scaladocLines(associatedDef)
+        // add `* @return` line only if the definition is a method def.
+        val returnLine =
+          if (lines.isEmpty && clientSupportsSnippets) "* @return $0"
+          else "* @return"
+        val newText =
+          if (associatedDef.isInstanceOf[DefDef])
+            (Seq("", "*") ++ lines ++ Seq(returnLine, "*/")).mkString("\n  ")
+          else (Seq("", "*") ++ lines ++ Seq("*/")).mkString("\n  ")
+        List(
+          new TextEditMember(
+            "Scaladoc Comment",
+            new l.TextEdit(
+              editRange,
+              newText
+            ),
+            completionsSymbol(associatedDef.name.toString()),
+            label = Some("/** */"),
+            detail = Some("Scaladoc Comment")
+          )
+        )
+      }
+    }
+
+    /**
+     * Returns the parameter lines of scaladoc besed on the given memberDef
+     * like ["* @param param1 $0", "* @param param2"].
+     *
+     * @param memberDef The memberDef to construct scaladoc.
+     */
+    private def scaladocLines(memberDef: MemberDef): List[String] = {
+      memberDef match {
+        case DefDef(_, _, _, vparamss, _, _) =>
+          vparamss.flatten.zipWithIndex
+            .map {
+              case (valdef, idx) => {
+                // /**
+                //   * @param param1 | <- move cursor to here.
+                //   * @param param2
+                //   */
+                if (idx == 0 && clientSupportsSnippets)
+                  s"* @param ${valdef.name} $$0"
+                else s"* @param ${valdef.name}"
+              }
+            }
+        case clazz @ ClassDef(_, _, _, _) =>
+          // If the associated def is a class definition,
+          // retrieve the constructor from the class, and caluculate the lines
+          // from the constructor definition instead.
+          new ConstructorFinder(clazz).getConstructor match {
+            case Some(defdef) => this.scaladocLines(defdef)
+            case scala.None => Nil
+          }
+        case _ => Nil
+      }
+    }
+
+    class ConstructorFinder(clazz: ClassDef) extends Traverser {
+      def getConstructor: Option[DefDef] = {
+        this.constructor = scala.None
+        clazz.impl.body.foreach(traverse)
+        constructor
+      }
+      private var constructor: Option[DefDef] = scala.None
+      override def traverse(tree: Tree): Unit = {
+        tree match {
+          case constructor @ DefDef(_, name, _, _, _, _)
```

Comment:
similar suggestion, it's a bit easier to read by doing:

```suggestion
          case constructor: DefDef
```

and then

```scala
if constructor.name == termNames.CONSTRUCTOR
```